packages/truffle-deployer/src/deployment.js

```diff
@@ -0,0 +1,404 @@
+        try {
+          eventArgs.estimate = await contract.new.estimateGas.apply(contract, newArgs);
+        } catch(err){
+          eventArgs.estimateError = err;
+        }
+
+        // Emit `preDeploy` & send transaction
+        await self.emitter.emit('preDeploy', eventArgs);
+        const promiEvent = contract.new.apply(contract, newArgs);
+
+        // Track emitters for cleanup on exit
+        self.promiEventEmitters.push(promiEvent);
+
+        // Subscribe to contract events / rebroadcast them to any reporters
+        promiEvent
+          .on('transactionHash', self._hashCb.bind(promiEvent, self, state))
+          .on('receipt',         self._receiptCb.bind(promiEvent, self, state))
+
+        await self._startBlockPolling(contract.web3);
+
+        // Get instance (or error)
+        try {
+          instance = await promiEvent;
+          self._stopBlockPolling();
+
+        } catch(err){
+
+          self._stopBlockPolling();
+          eventArgs.error = err.error || err;
+          await self.emitter.emit('deployFailed', eventArgs);
+          self._close();
+          throw new Error(self._errors(contract.contractName));
+        }
+
+      // Case: already deployed
+      } else {
+        instance = await contract.deployed();
+      }
+
+      // Emit `postDeploy`
+      eventArgs = {
+        contract: contract,
+        instance: instance,
+        deployed: shouldDeploy,
+        receipt: state.receipt
+      }
+
+      await self.emitter.emit('postDeploy', eventArgs);
+
+      // Wait for `n` blocks
+      if(self.confirmations !== 0 && shouldDeploy){
+        await self._waitBlocks(self.confirmations, state, contract.web3);
+      }
+      // Finish: Ensure the address and tx-hash are set on the contract.
+      contract.address = instance.address;
+      contract.transactionHash = instance.transactionHash;
+      return instance;
+    };
+  }
+
+  /**
+   * Deploys an array of contracts
+   * @param  {Array} arr  Array of contract abstractions to deploy
+   * @return {Promise}
+   */
+  _deployMany(arr){
+    const self = this;
+
+    return async function() {
+      const deployments = arr.map(args => {
+        let params;
+        let contract;
+
+        if (Array.isArray(args)) {
+          contract = args[0];
+
+          (args.length > 1)
+            ? params = args.slice(1)
+            : params = [];
+
+        } else {
+          contract = args;
+          params = [];
+        }
+
+        return self._deploy(contract, params)();
+      });
+
+      await self.emitter.emit('preDeployMany', arr);
+      await Promise.all(deployments);
+      await self.emitter.emit('postDeployMany', arr);
+    };
+  }
+
+  /**
+   * Cleans up promiEvents' emitter listeners
+   */
+  _close(){
+    this.emitter.clearListeners();
+    this.promiEventEmitters.forEach(item => {
+      item.removeAllListeners();
+    });
+  }
+};
+
+module.exports = Deployment;
```

Comment:
No new line at end of file

For those of us old-heads that like to `cat` their files on the terminal

Reply:
But really I came here to say that `_methods` should **not** be called outside of this class

Reply:
Fixed